addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -112,13 +115,21 @@ def _l10n_es_tbai_post_invoice_edi(self, invoice):
             error_msg = ''
             if chain_head and chain_head != invoice and not chain_head._l10n_es_tbai_is_in_chain():
                 error_msg = _("TicketBAI: Cannot post invoice while chain head (%s) has not been posted", chain_head.name)
-            if (
-                invoice.move_type == 'out_refund'
-                and not invoice.reversed_entry_id or (
-                    not invoice.reversed_entry_id._l10n_es_tbai_is_in_chain()
-                    and invoice.reversed_entry_id.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'es_tbai'))  # avoid imported ones
-            ):
-                error_msg = _("TicketBAI: Cannot post a reversal move while the source document (%s) has not been posted", invoice.reversed_entry_id.name)
+            refunded_invoices = self._l10n_es_tbai_refunded_invoices(invoice)
+            if invoice.move_type == 'out_refund':
+                refund_valid = refunded_invoices
+                for refunded_invoice in refunded_invoices:
+                    refund_valid = (
+                        refund_valid
+                        and refunded_invoice._l10n_es_tbai_is_in_chain()
+                        and refunded_invoice.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'es_tbai')  # avoid imported ones
+                    )
+                    if not refund_valid:
+                        error_msg = _(
+                            "TicketBAI: Cannot post a reversal move while the source document (%s) has not been posted",
+                            refunded_invoice.name
+                        )
+                        break
```

Comment:
I feel like there are useless variables and checks, I think this could be simplified to this, can't it ?

```suggestion
            if invoice.move_type == 'out_refund':
                refunded_invoices = self._l10n_es_tbai_refunded_invoices(invoice)
                for refunded_invoice in refunded_invoices:
                    if not (
                        refunded_invoice._l10n_es_tbai_is_in_chain()
                        and refunded_invoice.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'es_tbai')  # avoid imported ones
                    ):
                        error_msg = ...
                        break
```

Reply:
After reviewing, I saw that neither your code nor mine would fail in the case a refund doesn't indicate any refund invoices.

I updated it taking into account more or less your suggestion. Could you please review again?